### ISSUE TITLE
Incomplete Fix to "corrupt SD card" bug

### ISSUE THREAD

**Issue by robertzak133** [robertzak133/unified-btc-reverse#5]
More extensive field testing of current firmware version (2024-01-xx) shows that it can still fail.  This due to the fact that the fix only knocks down SD clock speed by one "tick".  For SD's that are merely 2x faster than what camera can support (like those I tested the fix on), this works great.  However, for faster SD cards (typical for modern, large capacity parts), the speed may have to be bumped down several times to get below 100 MHz (50 MHz).

**Comment by robertzak133** [comment]
the solution here it to create a loop which downgrades SD card speed until the part is operating slow enough for the camera.  An easy fix, to be released in next major version

**Comment by robertzak133** [comment]
on further inspection -- the original fix did indeed always reduced SD Clock to 50 MHz, so it only needed to be called once, as it was.  Which does not explain the re-occurence of the bug despite this fix.  I did notice that there was a window in which the SD card clock was set much higher (by default).  I added a new patch which over-rides the default when setting the SD clock so that it initializes to 50 MHz.  If corrupt sd card bug persists after that, I have some more serious debugging to do :(  This latest fix coming in next release

**Comment by robertzak133** [comment]
With the help of my trusty USB logic analyzer, and a couple of 200 MB/s rated 128GB cards, I was able to show that there were code paths which still allowed the SD card to operate beyond the safe (and perfectly adequate) 25 MHz.   With this as a guide, I made a systematic survey of all the places in the factory firmware where the sd clock is set beyond 25 MHz, and patched them out.  Verified fix with logic analyzer.  This should do it.
